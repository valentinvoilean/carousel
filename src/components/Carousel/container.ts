import { compose } from 'recompose';

import { Carousel } from './Carousel';

import { withButtonHandlers } from './withButtonHandlers';
import { withMouseHandlers } from './withMouseHandlers';
import { withReduxConnect } from './withReduxConnect';
import { withSlideHandlers } from './withSlideHandlers';
import { withTouchHandlers } from './withTouchHandlers';

import { ComposedProps, OwnProps } from './types';

export const CarouselContainer = compose<ComposedProps, OwnProps>(
  withReduxConnect,
  withSlideHandlers,
  withButtonHandlers,
  withMouseHandlers,
  withTouchHandlers
)(Carousel);
