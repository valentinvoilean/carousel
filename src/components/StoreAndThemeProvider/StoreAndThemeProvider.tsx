import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import configureStore from '../../store';
import { theme } from '../../theme';

export const StoreAndThemeProvider: React.FunctionComponent = ({ children }) => (
  <Provider store={configureStore()}>
    <ThemeProvider theme={theme}>
      <>{children}</>
    </ThemeProvider>
  </Provider>
);
