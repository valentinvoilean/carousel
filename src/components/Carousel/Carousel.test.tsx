import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

import { StoreAndThemeProvider } from '../StoreAndThemeProvider';
import { CarouselContainer as Carousel } from './container';

describe('Carousel', () => {
  it('renders without crashing', () => {
    const header = renderer
      .create(
        <StoreAndThemeProvider>
          <Carousel />
        </StoreAndThemeProvider>
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
