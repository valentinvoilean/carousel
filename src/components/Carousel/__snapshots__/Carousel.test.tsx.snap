// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders without crashing 1`] = `
.c0 {
  position: relative;
  width: 800px;
  overflow: hidden;
  box-shadow: 0px 0px 24px 0px rgba(0,0,0,0.35);
  background: #fff;
}

.c0::before {
  display: block;
  content: '';
  width: 100%;
  padding-top: calc(100% / (16/9));
}

.c1 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 200%;
  -webkit-transition: -webkit-transform 400ms;
  -webkit-transition: transform 400ms;
  transition: transform 400ms;
}

.c2 * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 img {
  width: 100%;
  height: 100%;
}

.c5 {
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background: url(arrow.svg) no-repeat;
  outline: none;
  border: 0;
  cursor: pointer;
  width: 50px;
  height: 50px;
}

.c4 {
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background: url(arrow.svg) no-repeat;
  outline: none;
  border: 0;
  cursor: pointer;
  width: 50px;
  height: 50px;
  left: 10px;
  right: auto;
  -webkit-transform: translateY(-50%) rotate(180deg);
  -ms-transform: translateY(-50%) rotate(180deg);
  transform: translateY(-50%) rotate(180deg);
}

.c6 {
  position: absolute;
  bottom: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  padding: 5px 10px;
  background: rgba(0,0,0,0.5);
  color: white;
  font-family: 'Raleway';
}

<div
  className="c0"
>
  <div
    className="c1"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="c2"
      style={
        Object {
          "transform": "translate3d(-50%, 0, 0)",
        }
      }
    >
      <div
        className="c3"
      />
      <div
        className="c3"
      />
    </div>
  </div>
  <button
    className="c4"
    onClick={[Function]}
  />
  <button
    className="c5"
    onClick={[Function]}
  />
  <div
    className="c6"
  >
    1
    /
    0
  </div>
</div>
`;
